framework/TSA/LinearRegression.py

```diff
@@ -0,0 +1,128 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Linear and Polynomial Regression
+"""
+import numpy as np
+import utils.importerUtils
+statsmodels = utils.importerUtils.importModuleLazy("statsmodels", globals())
+
+from utils import InputData, InputTypes, randomUtils, xmlUtils, mathUtils, utils
+from .TimeSeriesAnalyzer import TimeSeriesAnalyzer
+
+
+class LinearRegression(TimeSeriesAnalyzer):
+  """
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    specs = super(LinearRegression, cls).getInputSpecification()
+    specs.name = 'regression'
+    specs.description = """TimeSeriesAnalysis algorithm for fitting data of degree one or greater."""
+    specs.addSub(InputData.parameterInputFactory('degree', contentType=InputTypes.IntegerType,
+                                                 descr="Specifies the degree polynomial to fit the data with."))
+    return specs
+
+  #
+  # API Methods
+  #
+  def __init__(self, *args, **kwargs):
+    """
+      A constructor that will appropriately intialize a supervised learning object
+      @ In, args, list, an arbitrary list of positional values
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    # general infrastructure
+    TimeSeriesAnalyzer.__init__(self, *args, **kwargs)
+
+  def handleInput(self, spec):
+    """
+      Reads user inputs into this object.
+      @ In, inp, InputData.InputParams, input specifications
+      @ Out, settings, dict, initialization settings for this algorithm
+    """
+    settings = TimeSeriesAnalyzer.handleInput(self, spec)
+    settings['degree'] = spec.findFirst('degree').value
+    return settings
+
+  def characterize(self, signal, pivot, targets, settings):
+    """
+      Determines the charactistics of the signal based on this algorithm.
+      @ In, signal, np.ndarray, time series with dims [time, target]
+      @ In, pivot, np.1darray, time-like parameter values
+      @ In, targets, list(str), names of targets in same order as signal
+      @ In, settings, dict, additional settings specific to this algorithm
+      @ Out, params, dict, characteristic parameters
+    """
+    from sklearn.preprocessing import PolynomialFeatures
+    import statsmodels.api as sm
+
+    params = {f'{target}': {'model': {}} for target in targets}
```

Comment:
this doesn't need changing, but just checking; is `f'{target}'` different than `target` assuming the target names from RAVEN are always strings?